osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -152,24 +151,7 @@ protected Track GetVirtualTrack(double emptyLength = 0)
         {
             const double excess_length = 1000;
 
-            var lastObject = Beatmap?.HitObjects.LastOrDefault();
-
-            double length;
-
-            switch (lastObject)
-            {
-                case null:
-                    length = emptyLength;
-                    break;
-
-                case IHasDuration endTime:
-                    length = endTime.EndTime + excess_length;
-                    break;
-
-                default:
-                    length = lastObject.StartTime + excess_length;
-                    break;
-            }
```

Comment:
As I said in Discord, I think we still want this. Maps that were imported prior to `BeatmapInfo.Length` being a thing don't have it set correctly.

I was only suggesting it to be used as the fallback value in `LoadTrack`:
https://github.com/ppy/osu/blob/4f928232eef760de3d84145383297999988b489c/osu.Game/Beatmaps/WorkingBeatmap.cs#L110

Reply:
As per my reply on discord, this only affects a handful of beatmaps (those with no audio track in the first place) so I don't think it's too important, but if you insist I'll look to add a migration to fill in any non-calculated beatmaps.

Reply:
Hmm I suppose that's true.